core/domain/stats_domain.py

```diff
@@ -63,13 +66,119 @@
 MAX_ANSWER_DETAILS_BYTE_SIZE = 10000
 
 
+class SubmittedAnswerDict(TypedDict, total=False):
+    """Dictionary representing the SubmittedAnswer object."""
+
+    answer: str
+    interaction_id: str
+    answer_group_index: int
+    rule_spec_index: int
+    classification_categorization: str
```

Comment:
Should this be typed more strictly, if there are only a small number of possible types?

Reply:
Done